src/native_loader.ts

```diff
@@ -48,6 +48,7 @@ async function loadFromCLI(
   let loader: esbuild.Loader;
   switch (module.mediaType) {
     case "JavaScript":
+    case "Mjs":
```

Comment:
You many want to also add `.mts` file support, https://www.typescriptlang.org/docs/handbook/esm-node.html#new-file-extensions